### DISCUSSION TITLE
How to config c++'s formatter style using clang-format without warnings notification?

### DISCUSSION THREAD

**Discussion by GuangDai** [LunarVim/LunarVim#2539] [Q&A]
I config like below

local formatters = require "lvim.lsp.null-ls.formatters"

formatters.setup {
  { command = "clang-format",
    extra_args = { "--style=GNU" },
  },
}

but I get warnings when I open *.cpp files:
warning: multiple different client offset_encodings detected for buffer, this is not supported yet.

It's very annoying. Any way to fix it?

**Comment by GuangDai** [comment]
https://github.com/jose-elias-alvarez/null-ls.nvim/issues/428 Related issue from upstream. It mentions an interim "solution".

**Reply by kdamaski** [reply]
Hello may i ask you if you got the extra args to work?
I have tried different formats like the one below but none worked for me with the lunarvim file tree. I also checked in the runtime dir tree and didn't see any example of how to use formatter's extra args. If i put those args (without hte --style) in a .clang-format file in a working directory then it works great. Thanks in advance
```lua
		extra_args = { '--style=',
			'AllowShortFunctionsOnASingleLine: false',
			'AllowShortIfStatementsOnASingleLine: false',
			'AllowShortLoopsOnASingleLine: false',
			'IndentCaseLabels: false',
			'SortIncludes: false',
			'columnLimit: 103',
			'IndentWidth: 4',
		},
```

**Reply by GuangDai** [reply]
Sorry, I turn to emacs now. I meet a problem like yours in emacs. And most of my friends recommend me to use .clang file to format. So I follow their advice.
